scico/admm.py

```diff
@@ -210,9 +210,6 @@ def compute_rhs(self) -> Union[JaxArray, BlockArray]:
             if isinstance(self.admm.f, WeightedSquaredL2Loss):
```

Comment:
The `else` clause has already been discussed, but is there any point to this test either? The `__init__` raises an exception if `f` is not quadratic. Does that not guarantee that this test passes with the new class structure?

Reply:
Good point. I think we can remove that check.

Also, in 
https://github.com/lanl/scico/blob/a9db83098514dbca6bcdad8fc2e8e229ca37c988/scico/admm.py#L169-L177
we may rather want to test whether 
- A is instance of linop
- f is instance of WeightedSquaredL2Loss
(which implies it is quadratic)

I think it is conceptually better because all we want is to be able to evaluate
```python
         ATWy = self.admm.f.A.adj(self.admm.f.W.diagonal * self.admm.f.y) 
         rhs += 2.0 * self.admm.f.scale * ATWy 
```

Because technically we **_don't_** handle any case where
- f is not instance of WeightedSquaredL2Loss
- f is quadratic

Reply:
I'm inclined to agree. Is there even a good reason to keep the `is_quadratic` attribute?

Reply:
No. That part of the admm.py is the only place it is being used. I'll take care of both things.

Reply:
7455f1a

Reply:
I can try and update the test case to check for the instance of weighted to get the coverage up.

Reply:
IMO adding a test for when `not(f is None or f is WeightedSquaredL2Loss)` would not be very useful.

@bwohlberg do we have a policy that we desire every branch of execution to be covered by the tests?

Reply:
That is best practice.